app/models/adjustment.rb

```diff
@@ -0,0 +1,8 @@
+class Adjustment < ApplicationRecord
+  belongs_to :statement
+
+  validates :description, presence: true
+  validates :amount, presence: true
+  validates :amount, numericality: { other_than: 0 }, if: :amount
+  validates :amount, format: { with: /\A-?\d*\z/ }, if: -> { amount.present? }
```

Comment:
typecasting will convert this to the appropriate number (or a 0 if its a word) so could this also just be `if: :amount` ?

if the user types something invalid wont it just convert it to 0 and it'll hit the non-zero validation on line 6?

Reply:
I think the attribute is only typecast when it is saved.
```
> a = Adjustment.new(description:'test', amount:'test')
> a.valid?
=> false
> a.errors.messages
=> => {:statement=>["must exist"], :amount=>["You can only enter numeric values in the adjustment amount field"]}
```